### ISSUE TITLE
File.expand_path produces unexpected result with file protocol

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor.js#382]
Elia made some changes to make `File.expand_path` compliant to the Ruby spec: https://github.com/opal/opal/commit/838d0700d78c4f06f1513f555937ac30c495ddf2

Unfortunately it produces regressions on Asciidoctor.js because we are using a `File.expand_path` on a path starting with `file://`, more specifically:

https://github.com/asciidoctor/asciidoctor/blob/d6608c90384ce425df68972c423ed0780371a9f5/lib/asciidoctor/document.rb#L325

**before.js**
```js
Opal.const_get_qualified('::', 'File').$expand_path("file:///path/to/asciidoctor.js"); 
// file:///path/to/asciidoctor.js
```

**after.js**
```js
Opal.const_get_qualified('::', 'File').$expand_path("file:///path/to/asciidoctor.js"); 
// ./file:/path/to/asciidoctor.js
```

**mri.rb**
```rb
p File.expand_path('file:///path/to/asciidoctor.js')
# /path/to/file:/path/to/asciidoctor.js
```

I'm a bit surprised with MRI, I was expecting `file:///` to be considered an absolute path... :neutral_face: 

As a workaround we could define a path without the `file` protocol in the tests but then we will need to fix this issue : https://github.com/asciidoctor/asciidoctor/issues/2375 otherwise local path without protocol cannot be read when running in a browser environment. 

https://github.com/asciidoctor/asciidoctor.js/blob/54bc65cb74c922d8b714c2f16a64f52857091d1e/spec/browser/asciidoctor.spec.js#L5

If we don't want to fix https://github.com/asciidoctor/asciidoctor/issues/2375 right away, we could monkey-patch `::File.expand_path` in Asciidoctor.js to handle path starting with `file:///` ?

@mojavelinux I would like to have your opinion on this issue (short term and long term fix), thanks :stuck_out_tongue_winking_eye:

**Comment by mojavelinux** [comment]
This is being addressed upstream. In the long term, we will support a URI as a base dir. In the short term, we'll have an exception for the xmlhttprequest IO module.

**Comment by Mogztter** [comment]
Resolved in 10c1ca2
